src/librustdoc/clean/types.rs

```diff
@@ -676,15 +676,23 @@ impl Item {
         }
         let header = match *self.kind {
             ItemKind::ForeignFunctionItem(_) => {
-                let abi = tcx.fn_sig(self.item_id.as_def_id().unwrap()).abi();
+                let def_id = self.item_id.as_def_id().unwrap();
+                let abi = tcx.fn_sig(def_id).abi();
                 hir::FnHeader {
                     unsafety: if abi == Abi::RustIntrinsic {
                         intrinsic_operation_unsafety(tcx, self.item_id.as_def_id().unwrap())
                     } else {
                         hir::Unsafety::Unsafe
                     },
                     abi,
-                    constness: hir::Constness::NotConst,
+                    constness: if abi == Abi::RustIntrinsic
+                        && tcx.is_const_fn(def_id)
```

Comment:
Not sure if `"rust-intrinsic"` can ever be `const fn` or if they can only be `const` via `rustc_const_stable`

Reply:
Coming back to this, `is_const_fn` must be true since it's working in a chain of `&&`. It looks like `is_const_fn` takes account of raw const functions and const stable ones.

Reply:
We do have some const intrinsics like [`transmute`](https://doc.rust-lang.org/nightly/std/intrinsics/fn.transmute.html).